File: deploy.py
#!/usr/bin/env python3
"""
Deployment script for Alexa Discogs Skill (Python)
"""

import os
import shutil
import subprocess
import zipfile
from pathlib import Path

def create_deployment_package():
    """Create deployment package for AWS Lambda"""
    print("🎵 Creating Alexa Discogs Music Explorer deployment package...")
    
    # Clean up previous deployment
    if os.path.exists('deployment'):
        shutil.rmtree('deployment')
    
    if os.path.exists('alexa-discogs-skill-python.zip'):
        os.remove('alexa-discogs-skill-python.zip')
    
    # Create deployment directory
    os.makedirs('deployment')
    
    # Copy source files
    print("📦 Copying source files...")
    shutil.copytree('src', 'deployment/src')
    shutil.copy('lambda_function.py', 'deployment/')
    shutil.copy('requirements.txt', 'deployment/')
    
    # Install dependencies
    print("📦 Installing dependencies...")
    subprocess.run([
        'pip', 'install', '-r', 'requirements.txt', '-t', 'deployment/'
    ], check=True)
    
    # Create ZIP file
    print("📦 Creating ZIP package...")
    with zipfile.ZipFile('alexa-discogs-skill-python.zip', 'w', zipfile.ZIP_DEFLATED) as zipf:
        for root, dirs, files in os.walk('deployment'):
            for file in files:
                file_path = os.path.join(root, file)
                arcname = os.path.relpath(file_path, 'deployment')
                zipf.write(file_path, arcname)
    
    # Clean up deployment directory
    shutil.rmtree('deployment')
    
    print("✅ Deployment package created: alexa-discogs-skill-python.zip")
    print("")
    print("🎯 Next steps:")
    print("1. Upload the ZIP file to AWS Lambda")
    print("2. Set the handler to: lambda_function.handler")
    print("3. Set the runtime to: Python 3.9 or later")
    print("4. Configure Alexa Skills Kit trigger")
    print("5. Test your skill in the Alexa Developer Console")

if __name__ == "__main__":
    create_deployment_package()

File: deploy.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');
const archiver = require('archiver');

console.log('🎵 Creating Alexa Discogs Music Explorer deployment package (Node.js)...');

// Clean up previous deployment
if (fs.existsSync('deployment')) {
    fs.rmSync('deployment', { recursive: true, force: true });
}

if (fs.existsSync('alexa-discogs-skill-nodejs.zip')) {
    fs.unlinkSync('alexa-discogs-skill-nodejs.zip');
}

// Create deployment directory
fs.mkdirSync('deployment');

console.log('📦 Copying source files...');

// Copy source files
const filesToCopy = [
    'index.js',
    'package.json',
    'src'
];

filesToCopy.forEach(file => {
    const srcPath = path.join(__dirname, file);
    const destPath = path.join(__dirname, 'deployment', file);
    
    if (fs.statSync(srcPath).isDirectory()) {
        fs.cpSync(srcPath, destPath, { recursive: true });
    } else {
        fs.copyFileSync(srcPath, destPath);
    }
});

// Install production dependencies
console.log('📦 Installing production dependencies...');
try {
    execSync('npm install --production', { 
        cwd: path.join(__dirname, 'deployment'),
        stdio: 'inherit'
    });
} catch (error) {
    console.error('Failed to install dependencies:', error.message);
    process.exit(1);
}

// Create ZIP file
console.log('📦 Creating ZIP package...');
const output = fs.createWriteStream('alexa-discogs-skill-nodejs.zip');
const archive = archiver('zip', { zlib: { level: 9 } });

output.on('close', () => {
    console.log(`✅ Deployment package created: alexa-discogs-skill-nodejs.zip (${archive.pointer()} bytes)`);
    
    // Clean up deployment directory
    fs.rmSync('deployment', { recursive: true, force: true });
    
    console.log('');
    console.log('🎯 Next steps:');
    console.log('1. Upload the ZIP file to AWS Lambda');
    console.log('2. Set the handler to: index.handler');
    console.log('3. Set the runtime to: Node.js 18.x or later');
    console.log('4. Configure Alexa Skills Kit trigger');
    console.log('5. Test your skill in the Alexa Developer Console');
});

archive.on('error', (err) => {
    console.error('Error creating ZIP file:', err);
    process.exit(1);
});

archive.pipe(output);
archive.directory('deployment/', false);
archive.finalize();